src/Steps/Release/PublishRelease.php

```diff
@@ -77,43 +77,27 @@ protected function releaseLibrary(OutputInterface $output, LibraryRelease $relea
         $versionName = $releasePlanNode->getVersion()->getValue();
         $this->log($output, "Releasing library <info>{$name}</info> at version <info>{$versionName}</info>");
 
-        if (!is_null($branch)) { // skip if it's already detached
-            // Step 1: Push development branch to origin before tagging
-            $library->pushTo('origin');
+        // Step 1: Rewrite composer.json to all tagged versions only
+        $this->stabiliseRequirements($output, $releasePlanNode);
 
-            // Step 2: Detach head from current branch before modifying
-            $this->detachBranch($output, $library);
-        }
+        // Step 2: Push development branch to origin
+        $this->log($output, "Pushing branch <info>{$branch}</info>");
+        $library->pushTo('origin');
 
-        // Step 3: Rewrite composer.json on this head to all tagged versions only
-        $this->stabiliseRequirements($output, $releasePlanNode);
+        // Step 3. Make sure the patch release branch is pushed up to origin
+        $patchReleaseBranch = str_replace(RewriteReleaseBranches::RELEASE_SUFFIX, '', $branch);
+        if ($patchReleaseBranch !== $branch) {
```

Comment:
This was from before I added a check against this earlier in the process. Will remove.

Reply:
The release manager may have been doing something that required them to swap branches, and forgot to swap back. Probably it would be better to just bail in that case though, and tell the user to resolve the problem.
I'll do that instead.